### PR TITLE
Remove yet more iterator boilerplate

### PR DESCRIPTION
**Pull request by jwpeterson**
Thankfully, I think these patches are getting smaller over time. Keeping this split up into separate commits since just yesterday @roystgnr found an issue with one of these refactorings in the --disable-amr config. I'll be attaching multiple CIVET recipes to this one... 😄

### CODE REVIEW

**Comment by jwpeterson** [comment]
The Test debug failure is due to a recent CIVET configuration change, not a real issue. Will be invalidating shortly.

File: src/reduced_basis/rb_scm_evaluation.C [hunk 1]
```diff
@@ -534,17 +530,13 @@ void RBSCMEvaluation::legacy_read_offline_data_from_files(const std::string & di
...
-        {
-          std::string param_name = it->first;
-          Real param_value;
-          C_J_in >> param_value;
-          C_J[i].set_value(param_name, param_value);
-        }
-    }
+    for (const auto & pr : get_parameters())
+      {
+        std::string param_name = pr.first;
```
Comment (roystgnr) [inline src/reduced_basis/rb_scm_evaluation.C]: Should this be another one we change to const reference while we're at it?

Comment (jwpeterson) [reply src/reduced_basis/rb_scm_evaluation.C]: Yep, good catch.

Comment (jwpeterson) [reply src/reduced_basis/rb_scm_evaluation.C]: Oh no, wait a minute. The problem here was that `C_J_in` in this code chunk does not have an `>>` overload for const references if I recall correctly?

Comment (jwpeterson) [reply src/reduced_basis/rb_scm_evaluation.C]: Ah, misread similarly named parameters. I will change that.

**Pull request merged**